front/lib/auth.ts

```diff
@@ -334,30 +380,9 @@ export async function getUserFromSession(
     email: user.email,
     name: user.name,
     image: session.user ? session.user.image : null,
-    workspaces: workspaces.map((w) => {
-      const m = memberships.find((m) => m.workspaceId === w.id);
-      let role = "none" as RoleType;
-      if (m) {
-        switch (m.role) {
-          case "admin":
-          case "builder":
-          case "user":
-            role = m.role;
-            break;
-          default:
-            role = "none";
-        }
-      }
-      return {
-        id: w.id,
-        sId: w.sId,
-        name: w.name,
-        allowedDomain: w.allowedDomain || null,
-        role,
-        plan: planForWorkspace(w),
-        upgradedAt: w.upgradedAt?.getTime() || null,
-      };
-    }),
+    workspaces: await Promise.all(
+      workspaces.map((workspace) => _getWorkspaceType(workspace))
```

Comment:
this will make N sequential SQL queries where it was 0 before.

Can we not give the plan here and avoid going to SQL?

Maybe we should make this array a list of workspace sId (they already differ from real workspace objects I blelieve?)